src/erlcloud_waf.erl

```diff
@@ -0,0 +1,1152 @@
+
+
+%%%------------------------------------------------------------------------------
+%%% GetSizeConstraintSet
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_GetSizeConstraintSet.html
+%%%------------------------------------------------------------------------------
+-spec get_size_constraint_set/1 ::
+        (SizeConstraintSetId :: string() | binary()) ->
+        waf_return_val().
+get_size_constraint_set(SizeConstraintSetId) ->
+    get_size_constraint_set(SizeConstraintSetId, default_config()).
+
+-spec get_size_constraint_set/2 ::
+        (SizeConstraintSetId :: string() | binary(),
+         Config :: aws_config()) ->
+        waf_return_val().
+get_size_constraint_set(SizeConstraintSetId, Config) ->
+    Json = [{<<"SizeConstraintSetId">>, to_binary(SizeConstraintSetId)}],
+    waf_request(Config, "AWSWAF_20150824.GetSizeConstraintSet", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% GetSqlInjectionMatchSet
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_GetSqlInjectionMatchSet.html
+%%%------------------------------------------------------------------------------
+-spec get_sql_injection_match_set/1 ::
+        (SqlInjectionMatchSetId :: string() | binary()) ->
+        waf_return_val().
+get_sql_injection_match_set(SqlInjectionMatchSetId) ->
+    get_sql_injection_match_set(SqlInjectionMatchSetId, default_config()).
+
+-spec get_sql_injection_match_set/2 ::
+        (SqlInjectionMatchSetId :: string() | binary(),
+         Config :: aws_config()) ->
+        waf_return_val().
+get_sql_injection_match_set(SqlInjectionMatchSetId, Config) ->
+    Json = [{<<"SqlInjectionMatchSetId">>, to_binary(SqlInjectionMatchSetId)}],
+    waf_request(Config, "AWSWAF_20150824.GetSqlInjectionMatchSet", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% GetWebACL 
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_GetWebACL.html
+%%%------------------------------------------------------------------------------
+-spec get_web_acl/1 ::
+        (WebACLId :: string() | binary()) ->
+        waf_return_val().
+get_web_acl(WebACLId) ->
+    get_web_acl(WebACLId, default_config()).
+
+-spec get_web_acl/2 ::
+        (WebACLId :: string() | binary(),
+         Config :: aws_config()) ->
+        waf_return_val().
+get_web_acl(WebACLId, Config) ->
+    Json = [{<<"WebACLId">>, to_binary(WebACLId)}],
+    waf_request(Config, "AWSWAF_20150824.GetWebACL", Json).
+
+
+%%%------------------------------------------------------------------------------
+%%% ListByteMatchSets
+%%
+%% http://docs.aws.amazon.com/waf/latest/APIReference/API_ListByteMatchSets.html
+%%%------------------------------------------------------------------------------
+-spec list_byte_match_sets/1 ::
+       (Limit :: pos_integer()) ->
+       waf_return_val().
+list_byte_match_sets(Limit) when is_number(Limit) ->
```

Comment:
i understand that AWS WAF API claims Limit as required=YES (mindblown) for list APIs but I would suggest the following api with options and we did for inspector(https://github.com/alertlogic/erlcloud/blob/master/src/erlcloud_inspector.erl#L538). 
also I would not be surprised if Limit is actually optional and there is a default value
```erlang
list_byte_match_sets() ->
    list_byte_match_sets([], default_config()).
list_byte_match_sets(Limit) when is_number(Limit) ->
   list_byte_match_sets([{<<"Limit">>, Limit}], default_config());
list_byte_match_sets(Options) when is_list(Options) ->
   list_byte_match_sets(Options, default_config());
list_byte_match_sets(Config) when is_record(Config, aws_config) ->
   list_byte_match_sets([], Config).
list_byte_match_sets(Options, Config) when is_list(Option), is_record(Config, aws_config) ->
   list_byte_match_sets(Options, Config).
```

do you find it reasonable @ptrakhtman ?